### PR TITLE
Sage300 Main Menu Tabs

### PR DESCRIPTION
**Pull request by anishfyle**

### CODE REVIEW

File: src/app/shared/components/menu/main-menu/main-menu.component.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  templateUrl: './main-menu.component.html',
+  styleUrls: ['./main-menu.component.scss']
+})
+export class MainMenuComponent implements OnInit {
+
+  @Input() activeModule: MenuItem;
+
+  @Input() modules: MenuItem[];
+
+  moreDropdown: null;
```
Comment (DhaaraniCIT) [inline src/app/shared/components/menu/main-menu/main-menu.component.ts:15]: why it is null, any specific reason, what is the work of moreDropdown variable

File: src/app/integrations/sage300/sage300-main/sage300-main.component.ts [hunk 2]
```diff
@@ -7,6 +8,13 @@ import { Component, OnInit } from '@angular/core';
 })
 export class Sage300MainComponent implements OnInit {
 
+  modules: MenuItem[] = [
+    {label: 'Dashboard', routerLink: '/integrations/sage300/sage300-main/sage300-dashboard'},
+    {label: 'Export Log', routerLink: '/integrations/sage300/sage300-main/sage300-export_log'},
```
Comment (DhaaraniCIT) [inline src/app/integrations/sage300/sage300-main/sage300-main.component.ts:13]: routeLink should be like `/integrations/sage300/main/dashboard`, no need to mention sage300 for main and other router, mensioning sage300 once after the integration/ is sufficient

File: src/app/integrations/sage300/sage300-main/sage300-main.component.ts [hunk 3]
```diff
@@ -7,6 +8,13 @@ import { Component, OnInit } from '@angular/core';
...
+    {label: 'Mapping', routerLink: '/integrations/sage300/sage300-main/sage300-mapping'},
+    {label: 'Configuration', routerLink: '/integrations/sage300/sage300-main/sage300-configuration'}
+  ];
```
Comment (DhaaraniCIT) [inline src/app/integrations/sage300/sage300-main/sage300-main.component.ts:16]: same as above

File: src/app/shared/components/menu/main-menu/main-menu.component.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<div class="tw-border-gray-300 tw-rounded tw-flex tw-items-center tw-justify-between">
+    <p-tabMenu [model]="modules" [activeItem]="activeModule" ></p-tabMenu>
+
+    <div class="tw-flex tw-items-center">
+        <button [pTooltip]="'The integration will import all the newly updated Sage Intacct dimensions and Fyle expenses in the configured state of export'"
```
Comment (ashwin1111) [inline src/app/shared/components/menu/main-menu/main-menu.component.html:5]: "Sage Intacct", this should be appName, get it as input

File: src/app/shared/components/menu/main-menu/main-menu.component.html [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            (click)="null"
```
Comment (ashwin1111) [inline src/app/shared/components/menu/main-menu/main-menu.component.html:6]: emit output()

File: src/app/shared/components/menu/main-menu/main-menu.component.html [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+            class="tw-h-32-px tw-w-80-px tw-mr-12-px tw-flex tw-items-center focus:tw-ring-0 tw-justify-center tw-rounded tw-bg-white tw-text-black tw-border tw-border-gray-300 focus:tw-outline-none focus:tw-ring-opacity-50 active:tw-scale-95 transition-transform">
+            <img src="assets/icons/stepper/sync.svg">
+            <span class="tw-text-14-px tw-pl-8-px">
+                Sync
+            </span>
+        </button>
+
+        <p-dropdown [(ngModel)]="moreDropdown" (onChange)="null" appendTo="body" class="tw-mr-26-px" [options]="['Submit Feedback']" placeholder="More">
```
Comment (ashwin1111) [inline src/app/shared/components/menu/main-menu/main-menu.component.html:14]: get a boolean flag as input to show / hide dropdown

File: src/app/shared/components/menu/main-menu/main-menu.component.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  templateUrl: './main-menu.component.html',
+  styleUrls: ['./main-menu.component.scss']
+})
+export class MainMenuComponent implements OnInit {
+
+  @Input() activeModule: MenuItem;
+
+  @Input() modules: MenuItem[];
+
+  @Input() moreDropdown = null;
```
Comment (ashwin1111) [inline src/app/shared/components/menu/main-menu/main-menu.component.ts:15]: `isDropdrownRequired`, boolean

File: src/app/shared/components/menu/main-menu/main-menu.component.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import { MenuItem } from 'primeng/api';
+
+@Component({
+  selector: 'app-main-menu',
+  templateUrl: './main-menu.component.html',
+  styleUrls: ['./main-menu.component.scss']
+})
+export class MainMenuComponent implements OnInit {
+
+  @Input() activeModule: MenuItem;
```
Comment (ashwin1111) [inline src/app/shared/components/menu/main-menu/main-menu.component.ts:11]: remove

File: src/app/shared/components/menu/main-menu/main-menu.component.html [hunk 9]
```diff
@@ -0,0 +1,23 @@
+<div class="tw-border-gray-300 tw-rounded tw-flex tw-items-center tw-justify-between">
+    <p-tabMenu [model]="modules" [activeItem]="activeModule" ></p-tabMenu>
```
Comment (ashwin1111) [inline src/app/shared/components/menu/main-menu/main-menu.component.html:2]: for activeItem, let's do `modules[0]`, let me know if it doesn't work

File: src/app/shared/components/menu/main-menu/main-menu.component.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  templateUrl: './main-menu.component.html',
+  styleUrls: ['./main-menu.component.scss']
+})
+export class MainMenuComponent implements OnInit {
+
+  @Input() modules: MenuItem[];
+
+  @Input() moreDropdown = null;
+
+  @Input() appName: string;
```
Comment (ashwin1111) [inline src/app/shared/components/menu/main-menu/main-menu.component.ts:15]: appName should be an enum

File: src/app/shared/components/menu/main-menu/main-menu.component.ts [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+  @Input() isDropdrownRequired: boolean;
+
+  @Output() refreshDimension = new EventEmitter<boolean>();
```
Comment (ashwin1111) [inline src/app/shared/components/menu/main-menu/main-menu.component.ts:19]: onRefreshDimensionClick

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**